presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -357,29 +359,148 @@ public void testQuantifiedComparisonNotEqualAny()
         );
     }
 
+    @Test
```

Comment:
Sure. As @martint said, he will merge the changes to optimizers first and then I will refactor those tests after your work is merged.